pkgs/build-support/node/build-npm-package/hooks/npm-config-hook.sh

```diff
@@ -0,0 +1,80 @@
+npmSetupHook() {
+    echo "Executing npmSetupHook"
+
+    runHook preConfigure
+
+    echo "Configuring npm"
+
+    export HOME=$TMPDIR
+    export npm_config_nodedir="@nodeSrc@"
+
+    npm config set cache "$npmDeps"
+    npm config set offline true
+    npm config set progress false
+
+    cacheLockfile="$npmDeps/package-lock.json"
+    srcLockfile="$NIX_BUILD_TOP/$sourceRoot/package-lock.json"
+
+    echo "Validating consistency between $srcLockfile and $cacheLockfile"
+
+    if ! @diff@ $srcLockfile $cacheLockfile; then
+      # If the diff failed, first double-check that the file exists, so we can
+      # give a friendlier error msg.
+      if ! [ -e $srcLockfile ]; then
+        echo "ERROR: Missing package-lock.json from src. Expected to find it at: $srcLockfile"
+        echo "Hint: You can use the patches attribute to add a package-lock.json manually to the build."
+        exit 1
+      fi
+
+      if ! [ -e $cacheLockfile ]; then
+        echo "ERROR: Missing lockfile from cache. Expected to find it at: $cacheLockfile"
+        exit 1
+      fi
+
+      echo
+      echo "ERROR: npmDepsHash is out of date"
+      echo
+      echo "package-lock.json is not the same in $npmDeps"
+      echo
+      echo "To fix the issue:"
+      echo '1. Use `lib.fakeHash` as the npmDepsHash value'
+      echo "2. Build the derivation and wait for it to fail with a hash mismatch"
+      echo "3. Copy the 'got: sha256-' value back into the npmDepsHash field"
+      echo
+
+      exit 1
+    fi
+
+    echo "Installing dependencies"
+
+    # Frozen install...
+    npm ci --ignore-scripts
+
+    # ...patch shebangs...
+    patchShebangs node_modules
+
+    root=$(pwd)
+
+    while IFS= read -r f; do
+        cd "$f" || continue
+
+        echo "$f"
```

Comment:
Seems like this only clogs up logs, the script that's being ran gets logged already anyways.

```suggestion
```